packages/core/src/wallet/metamaskBridge.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  DisconnectedStarknetWindowObject,
+  RpcMessage,
+} from "../StarknetWindowObject"
+import wallets, { WalletProvider } from "../discovery"
+import type { MetaMaskProvider } from "@consensys/get-starknet/dist/type"
+import detectEthereumProvider from "@metamask/detect-provider"
+import type { ProviderInterface } from "starknet"
+
+class EmptyMetaMaskProvider implements DisconnectedStarknetWindowObject {
+  id: string
+  name: string
+  icon: string
+  version = "0.0.0"
+  isConnected = false as const
+  provider: undefined | ProviderInterface
+  constructor(
+    private metamaskProvider: MetaMaskProvider,
+    walletInfo: WalletProvider,
+  ) {
+    this.id = walletInfo.id
+    this.name = walletInfo.name
+    this.icon = walletInfo.icon
+  }
+  request<T extends RpcMessage>(): Promise<T["result"]> {
+    throw new Error("Wallet not enabled")
+  }
+  async enable(options?: { starknetVersion?: "v4" | "v5" }): Promise<string[]> {
+    const { MetaMaskSnapWallet } = await import("@consensys/get-starknet")
+    const metaMaskSnapWallet = new MetaMaskSnapWallet(this.metamaskProvider)
+
+    Object.assign(this, metaMaskSnapWallet)
+    this.constructor.prototype = metaMaskSnapWallet.constructor.prototype
+
+    return await this.enable(options)
```

Comment:
await this.enable(options) may throw an error that the user reject to install the snap into their metamask, hence i would recommend

```suggestion

    try {
       const result = await metaMaskSnapWallet.enable(options)
       Object.assign(this, metaMaskSnapWallet)
       this.constructor.prototype = metaMaskSnapWallet.constructor.prototype
       return result
     } catch(e) {
       // do nothing or log the error
     }

    
```